app/models/bike.rb

```diff
@@ -75,8 +75,10 @@ class Bike < ActiveRecord::Base
   }
   scope :stolen, -> { where(stolen: true) }
   scope :non_stolen, -> { where(stolen: false) }
+  scope :abandoned, -> { where(abandoned: true) }
   scope :organized, -> { where.not(creation_organization_id: nil) }
   scope :with_known_serial, -> { where.not(serial_number: "unknown") }
+  scope :with_known_normalized_serial, -> { where.not(serial_normalized: %w[absent N0NE]) }
```

Comment:
This scope shouldn't be added, it only seems necessary because of incompletely migrated data (from #988). Rather than adding this scope, we should just fix the data:

- `absent` is the legacy version of `made_without_serial` - the serial_normalized fields that contain absent should be blanked out.
- Bikes with `none` should be switched to the current pattern - which is `serial_number: "made_without_serial",  made_without_serial: true` (it looks like there haven't been any new bikes added with "none" since the #988 shipped, so I think that update fixed that problem).

Reply:
Thanks for clarifying!

Just to check my understanding: 

If a bike record's 

1. `serial_number` is any of `absent` `none` or `N0NE` (but not if it's `"unknown"`), or its
2. `serial_normalized` is any of `absent` `none` or `N0NE`

then the record needs to be migrated to have the following:

```rb
{
  made_without_serial: true
  serial_number: "made_without_serial"
  serial_normalized: nil
}
```

Is that correct?



Just want to make sure I understand correctly, because this will migrate about ~12% of the bikes table:

```rb
Bike.where(serial_number: ["absent", "none", "N0NE"]).count
# => 27
Bike.where(serial_normalized: ["absent", "none", "N0NE"]).count
# => 32032
```

Do we also expect a `serial_number` of `"unknown"` to have a blank `serial_normalized`?

```rb
Bike.where(serial_number: "unknown").where.not(serial_normalized: nil).count
# => 32189

Bike.where(serial_number: %w[unknown made_without_serial]).pluck(:serial_normalized).compact.uniq.sort

# => ["1 D0N T KN0W", "1 D0N T KN0W 1T", "1 D0NT KN0W", "1 D0NT KN0W 1T", "1 D0NT KN0W REMEM8ER", "180968", "1DK", "1N2N8143234", "48TC3329", "5AD1Y 1 D0N T KN0W", "5Q1862028", "C070143604", "CU5T0M", "CU5T0M 81KE", "CU5T0M 81KE N0 5ER1A1 HA5 A UN1QUE FRAME DE51GN", "CU5T0M 8U11T", "CU5T0M F100R 81KE", "CU5T0M FRAME", "CU5T0M FRAME N0 5ER1A1", "CU5T0M MADE", "D0 N0T KN0W", "D0N T KN0W", "D0N T KN0W 1T", "D0N T KN0W 5ER1A1", "D0N T KN0W THE 5ER1A1 NUM8ER", "D0NT KN0W", "D0NT KN0W 1T", "D0NT KN0W 1T 8UT R0CKY M0UNTA1N 5P0RT5 MA1NE 5T1CKER", "D0NT KN0W 1T YET WA1T1NG 0N 01D 0WNER", "F1 56 767", "F1T 111", "M0000000000N A", "M1551NG", "M1551NG 5ER1A1", "M1551NG 5ER1A1 NUM8ER", "N A", "N A CU5T0M 8U11T N0 5ER1A1", "N A DAK0TA CARVED 0N 80TT0M 8RACKET", "N AXXX", "N0 5ER1A1", "N0 5ER1A1 5PEC1A1 MADE", "N0 5ER1A1 HAND 8U11T 81KE", "N0 5ER1A1 NUM8ER", "N0 5ER1A1 NUM8ER CU5T0M 8U11D", "N0 5ER1A1 NUM8ER CU5T0M HEAD8ADGE", "N0NE CU5T0M", "N0NE CU5T0M HANDMADE 81KE", "PR08A81Y HA5 0NE D0N T KN0W 1T", "UNKN0WN", "YF006", "absent"]

```

I can add a callback to enforce all this — it looks like there's some other dirty / inconsistent fields:

```rb
Bike.where(serial_number: "made_without_serial").where.not(made_without_serial: true).count
=> 4

Bike.where(serial_number: "unknown").where.not(serial_normalized: nil).count
# => 32189
```